jupyter_server/services/sessions/sessionmanager.py

```diff
@@ -72,7 +76,7 @@ def __eq__(self, other: object) -> bool:
                 )
         return False
 
-    def update(self, other: "KernelSessionRecord") -> None:
+    def update(self, other: KernelSessionRecord) -> None:
```

Comment:
I found the previous forward reference with the string-based value to still be necessary and is one of the sources of the current CI failures.

Reply:
Hum, that is weird.

```suggestion
    def update(self, other: "KernelSessionRecord") -> None:
```

Reply:
Hum, let me try with `from __future__ import annotations`